### DISCUSSION TITLE
Explore integration with OpenLineage

### DISCUSSION THREAD

**Discussion by astrojuanlu** [kedro-org/kedro#4054] [Idea]
> OpenLineage enables consistent collection of lineage metadata, creating a deeper understanding of how data is produced and used.

https://openlineage.io/

![image](https://github.com/user-attachments/assets/3c95b22c-1758-4abb-a2b7-4ee8a05df889)

Reference implementation: Marquez https://github.com/MarquezProject/marquez

![Marquez lineage UI](https://marquezproject.ai/img/unified_graph.png)

Validation:

> With the release of Airflow 2.7.0, OpenLineage became a [built-in feature of Airflow](https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/index.html).

What does this mean for us? Should Kedro-Viz consume OpenLineage exports coming from Kedro? (See previous discussion at https://github.com/kedro-org/kedro-viz/discussions/1833#discussioncomment-8931728)

**Comment by astrojuanlu** [comment]
A user [asked](https://linen-slack.kedro.org/t/22726218/hi-team-i-m-currently-working-with-the-project-using-databri#fa5a8d09-04b5-492c-8c9c-ae0a7ea50465)

> Is there any way to show log in Databricks for running pipelines/nodes execution such as time spending to complete specific node, memory usage, or cost in graphical manner like the image below?

> I was just wondering there are any ways to visualize pipeline execution results/logs in graphical manner like for example, this node takes this much time to complete, etc.

Exporting OL data addresses this.

**Comment by astrojuanlu** [comment]
Another user:

> What we are trying to capture and track is column-level spark transformations across the whole kedro pipeline

(hinted that only Kedro-related lineage would likely be insufficient)

**Comment by astrojuanlu** [comment]
OpenLineage allows solutions like DataHub to support more integrations. For example

> DataHub's REST endpoint allows users to send OpenLineage events directly to DataHub.

https://datahubproject.io/docs/lineage/openlineage/

> To automatically extract lineage information, the v2 plugin builds on top of Airflow's built-in [OpenLineage extractors](https://openlineage.io/docs/integrations/airflow/default-extractors).

https://datahubproject.io/docs/lineage/airflow/#automatic-lineage-extraction

**Comment by astrojuanlu** [comment]
Well, here's a proof of concept!

https://github.com/astrojuanlu/kedro-openlineage

![image (29)](https://github.com/user-attachments/assets/23e0fc39-8153-45cc-bed4-f14d0c355608)

**Comment by astrojuanlu** [comment]
https://www.bearingnode.com/post/openlineage-vendor-integration-status

**Comment by astrojuanlu** [comment]
The "exploration" is done.

**Comment by astrojuanlu** [comment]
Now that we're working on visualising the run status and we plan to show live data https://github.com/kedro-org/kedro-viz/issues/2310, I think there's some overlap with OpenLineage. @jitu5 @Huongg @rashidakanchwala should we do a spike on this after the MVP is released?

**Reply by bparbhu** [reply]
We should bring this up again
